### PR TITLE
Unify the highly duplicated size scheduler tests and also half the runtime

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java [hunk 1]
```diff
@@ -16,26 +16,176 @@
...
+  @AfterClass
+  public static void tearDown() throws NotFoundException, NamespaceCannotBeDeletedException {
+    namespaceAdmin.deleteNamespace(Constants.DEFAULT_NAMESPACE_ID);
+  }
+
+  /**
+   * Waits until there is nothing running for the given program.
+   */
+  private void waitUntilFinished(final ProgramRuntimeService runtimeService,
+                                   final Id.Program program, long maxWaitSeconds) throws Exception {
```
Comment (bdmogal) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java]: indentation

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java [hunk 2]
```diff
@@ -16,26 +16,176 @@
...
+        return isProgramRunning(runtimeService, program);
+      }
+    }, maxWaitSeconds, TimeUnit.SECONDS, 50, TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * Waits for the given program ran for the given number of times.
+   */
+  private void waitForRuns(final Store store, final Id.Program programId,
+                             int expectedRuns, long timeoutSeconds) throws Exception {
```
Comment (bdmogal) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java]: same here

**Comment by bdmogal** [comment]
@chtyim just a couple of minor checkstyle issues, otherwise LGTM. 

Thanks for this cleanup. I had seen this during #1624, but didn't do it to not divert attention from that PR. This fixes https://issues.cask.co/browse/CDAP-1807.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java (hunk 1 above)
Comment (chtyim) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java]: fixed.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java (hunk 2 above)
Comment (chtyim) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/SchedulerTestBase.java]: fixed.

**Comment by chtyim** [comment]
Build passed. http://builds.cask.co/browse/CDAP-DUT1247-1

**Pull request merged**